lib/chef/provider/aws_vpc.rb

```diff
@@ -160,37 +161,42 @@ def update_internet_gateway(vpc)
     end
   end
 
-  def update_main_route_table(vpc)
-    desired_route_table = Chef::Resource::AwsRouteTable.get_aws_object(new_resource.main_route_table, resource: new_resource)
+  def update_main_route_table(vpc, main_route_table)
+    desired_route_table = Chef::Resource::AwsRouteTable.get_aws_object(main_route_table, resource: new_resource)
     current_route_table = vpc.route_tables.main_route_table
     if current_route_table != desired_route_table
       main_association = current_route_table.associations.select { |a| a.main? }.first
       if !main_association
-        raise "No main route table association found for VPC #{new_resource.name} (#{vpc.id})'s current main route table #{current_route_table.id}: error!  Probably a race condition."
+        raise "No main route table association found for #{new_resource.to_s} current main route table #{current_route_table.id}: error!  Probably a race condition."
       end
-      converge_by "change main route table for VPC #{new_resource.name} (#{vpc.id}) to #{desired_route_table.id} (was #{current_route_table.id})" do
+      converge_by "change main route table for #{new_resource.to_s} to #{desired_route_table.id} (was #{current_route_table.id})" do
         vpc.client.replace_route_table_association(
           association_id: main_association.id,
-          route_table_id: desired_route_table.id)
+          route_table_id: desired_route_table.id
+        )
       end
     end
     desired_route_table
   end
 
-  def update_main_routes(vpc, main_route_table)
-    main_route_table ||= vpc.route_tables.main_route_table
-    aws_route_table main_route_table do
-      vpc vpc
-      routes new_resource.main_routes
+  # @param main_routes [Hash]
+  def update_main_routes(vpc, main_routes)
+    require 'pry'; binding.pry
+    ::Chef::Provisioning.inline_resource(action_handler) do
```

Comment:
@jkeiser this is what I'm having trouble with.  When I try to do this, I'm missing the `desired_driver` in the `aws_provider`.